pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1448,7 +1449,14 @@ public void addTable(TableConfig tableConfig)
          * We also need to support the case when a high-level consumer already exists for a table and we are adding
          * the low-level consumers.
          */
-        ensureRealtimeClusterIsSetUp(tableConfig);
+        try {
```

Comment:
Let's have a larger try-catch that covers both offline case and realtime case (having 2 separate try-catch is also fine. Essentially when the table creation didn't finish properly, we should clean up all the records already written to ZK